### DISCUSSION TITLE
Fluid Properties | Failed Compilation

### DISCUSSION THREAD

**Discussion by rl3fz** [idaholab/moose#18520] [Q&A Modules: General] [answered]
Hello everyone!

I tried to include the fluid properties module (switched from 'no' to 'yes' in my application's Makefile),
but when I tried to compile and test, it failed with the output: FAILED (Application not found)

Does anyone know how I can proceed? The application was working fine beforehand, and I updated MOOSE yesterday.

(Any and all advice appreciated!)

**Comment by GiudGiud** [comment, marked as answer]
That's odd

Did you run 'make clobberall' before rebuilding?

**Reply by rl3fz** [reply]
I had not, but I just tried that (along with make -j4 and ./run_tests -j4) and got the same result.

**Reply by GiudGiud** [reply]
Might need to git clean -xfd (after committing ALL your changes and saving ALL your work)
Could you please paste the the error message from the compilation
The FAILED error message just means an executable was not created

**Reply by rl3fz** [reply]
After all of the "Creating Unity..." lines, it showed the following:

/home/****/miniconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/9.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/****/miniconda3/envs/moose/lib/libgomp.so.1: undefined reference to `clock_gettime@GLIBC_2.17'
/home/****/miniconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/9.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/****/miniconda3/envs/moose/lib/libgomp.so.1: undefined reference to `clock_getres@GLIBC_2.17'
/home/****/miniconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/9.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/****/miniconda3/envs/moose/lib/libgomp.so.1: undefined reference to `aligned_alloc@GLIBC_2.16'
/home/****/miniconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/9.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/****/miniconda3/envs/moose/lib/libgomp.so.1: undefined reference to `secure_getenv@GLIBC_2.17'
collect2: error: ld returned 1 exit status
make: *** [/home/****/projects/moose/framework/moose.mk:374: /home/****/projects/moose/framework/contrib/exodiff/exodiff] Error 1
make: *** Waiting for unfinished jobs....

**Reply by GiudGiud** [reply]
if you switch fluid_properties back does it work again?
Can you compile in moose/framework/ ?

Did you update Conda?
These errors really look like you have previously compiled binaries and updated your Conda environment. Running `git clean -xfd` in your app and in moose should help.

**Reply by rl3fz** [reply]
> if you switch fluid_properties back does it work again?

No
> Can you compile in moose/framework/ ?

No

> Did you update Conda?

Not since Monday, but after updating today, I was able to compile moose/framework/ , and my application (without fluid properties) compiled, so I tried including fluid properties again, and it worked!

Seems I should be updating more frequently (or forgetting 'make clobberall' earlier really messed with the system), but I'll keep 'git clean -xfd' in mind for next time. Thank you!!

(edited for clarity)
